### PR TITLE
cpu/stm32_common: disable i2c in release

### PR DESCRIPTION
**Pull request by toonst**
### Contribution description
The hardware peripheral should be disabled before stopping the peripheral clock. With the uart driver we found that there could be some wrong behaviour otherwise.

### Testing procedure
use periph_i2c test app

### CODE REVIEW

File: cpu/stm32_common/periph/i2c_2.c [hunk 1]
```diff
@@ -186,23 +186,34 @@ int i2c_acquire(i2c_t dev)
     mutex_lock(&locks[dev]);
 
 #ifdef STM32_PM_STOP
-    /* block STOP mode */
```
Comment (aabadie) [inline cpu/stm32_common/periph/i2c_2.c]: Unrelated

File: cpu/stm32_common/periph/i2c_2.c [hunk 2]
```diff
@@ -186,23 +186,34 @@ int i2c_acquire(i2c_t dev)
...
+
+    if (_wait_for_bus(i2c) < 0) {
+        mutex_unlock(&locks[dev]);
+        return -ETIMEDOUT;
+    }
+
     periph_clk_dis(i2c_config[dev].bus, i2c_config[dev].rcc_mask);
 
 #ifdef STM32_PM_STOP
-    /* unblock STOP mode */
```
Comment (aabadie) [inline cpu/stm32_common/periph/i2c_2.c]: Unrelated change

File: cpu/stm32_common/periph/i2c_2.c [hunk 3]
```diff
@@ -186,23 +186,34 @@ int i2c_acquire(i2c_t dev)
...
+    i2c_config[dev].dev->CR1 |= I2C_CR1_PE;
+
     return 0;
 }
 
 int i2c_release(i2c_t dev)
 {
     assert(dev < I2C_NUMOF);
 
+    I2C_TypeDef *i2c = i2c_config[dev].dev;
```
Comment (aabadie) [inline cpu/stm32_common/periph/i2c_2.c]:
```suggestion
    /* disable device */
    i2c_config[dev].dev->CR1 &= ~(I2C_CR1_PE);
```

File: cpu/stm32_common/periph/i2c_1.c [hunk 4]
```diff
@@ -142,13 +142,26 @@ int i2c_acquire(i2c_t dev)
...
+    i2c_config[dev].dev->CR1 |= I2C_CR1_PE;
+
     return 0;
 }
 
 int i2c_release(i2c_t dev)
 {
     assert(dev < I2C_NUMOF);
 
+    I2C_TypeDef *i2c = i2c_config[dev].dev;
```
Comment (aabadie) [inline cpu/stm32_common/periph/i2c_1.c]:
```suggestion
    /* disable device */
    i2c_config[dev].dev->CR1 &= ~(I2C_CR1_PE);
```

**Comment by MrKevinWeiss** [comment]
[HiL tests](https://ci.riot-os.org/hil/blue/organizations/jenkins/RIOT-HIL/detail/master/23/pipeline/106) running

**Comment by MrKevinWeiss** [comment]
From what it seems the 
```
nucleo-f091rc
nucleo-f103rb
nucleo-f411re
nucleo-f767zi
nucleo-l073rz
```
all seem to be working as expected.

Since the behavior is undefined after release and this seems to help with the power mode I think can ACK.

**Comment by aabadie** [comment]
Thanks for the testing results @MrKevinWeiss !
@toonst can you update the PR following my suggestions ? You can squash directly.

**Comment by toonst** [comment]
@aabadie thanks for the review and @MrKevinWeiss thanks for testing. Unfortunately it seems this change is still causing some issues in our product. I'll hold it off for now till I completely understand what the issue is. I will update this PR when I have more info.

**Comment by MrKevinWeiss** [comment]
Maybe you can provide clearer test steps to reproduce the error?

**Comment by tcschmidt** [comment]
@toonst any new insights on the issues you encounter?

**Comment by toonst** [comment]
I haven't found the time yet, will try to get into this next week.

**Comment by kaspar030** [comment]
> I haven't found the time yet, will try to get into this next week.

@toonst *ping*

**Comment by aabadie** [comment]
@MrKevinWeiss, mind to give this one another round on the HiL ?

**Comment by MrKevinWeiss** [comment]
[Running](https://ci.riot-os.org/hil/blue/organizations/jenkins/RIOT-HIL/detail/master/3/pipeline)

It would be good to compare to the [nightly](https://ci.riot-os.org/hil/blue/organizations/jenkins/RIOT-HIL/detail/nightly/18/pipeline) for reference.

I will give a summary when it is done.

**Comment by MrKevinWeiss** [comment]
It appears no new regressions were introduced!
